### ISSUE TITLE
[PSET] Investigate Elements range proof encoding

### ISSUE THREAD

**Issue by altafan** [vulpemventures/go-elements#110]
The base64 string of a pset return by the elements CLI and including confidential inputs/outputs (in general, containing at least 1 range proof) is much shorter than the one produced with our pset package.

We have to investigate if and how range proofs are compressed in the encoding of the PSET.

This is an example of a PSET with 1 conf input and 2 conf outputs made with the CLI:

```
cHNldP8BAP0UAQIAAAAAAeeMNAx12ipWTC41v4fiM6ywKcNhS1fhQv7OsVHEQaZvAAAAAAD9////Aws+AhuKU4NouAPRZzFnDAoKQ8DaDX7lTNv+k34qfgpGMgmiKNnvrYseLDSmH+KVWXJCHZdAeIZYu/xCCDN6WvZKPgAWABSYjdann3nlHfJJtfXQDuY+aJ0snAoQkJXGdBL4fBBMAi8KVYlDzBBwtf/ajO5QS18EhQ2fMQl+d+Ai8D2curz0w7BYAAUt+FTfYRIDa7/sIeS7grZduQAWABTpXlsoaSHgqQOj2oUqlFdEjRIYiAElslEHDinKGQQ88zzNcyTi3asD7MSuC153xPwOXPbJWgEAAAAAAAASpAAAAAAAAAAAC/wIZWxlbWVudHMGIQKaffDQ1q04yYHUzgpuxvWkcE0DjDY9N8qStsOkvgdgHwAL/AhlbGVtZW50cwYhA9d0ZWZ+ZnmmSSe27BC26pRpEQIk3SAc2wYVeTqnumHPAAA=
```

While this is a pset with the same number of inputs and outputs created with go-elements:

```
cHNldP8BAP2HIgIAAAABAY3rACCsFaOMa9nKrA6oHYhaDpdKX8/QO91NLMGVMiXMAQAAAAD/////AwoiwQ5fZn2xGsBwyqcLaZpy7VRoLntt+E9y0T8lNA1U5gk3gkh08cTAIMY0/UvmtI3/fHRVt8rlGhScyWSAWUimBQL+3Y7JcTrgvOCCjH4jKBjrwgNbVbskIZ45wwSLF6F6ghl2qRQ5OXCAtR7yLFm9dGmvrP++7A2hLoisCiumZs8NRSw+JBt8+44DG3VvzMGhGlKDOTwRs1bS4rghCGCicquNjdBS6CGBxk8nwWPO1h1Fvb35780gmEzbGEb3AuvKhWLB1idbQNKzKsmZ9ztiP2WYHJuto9SFYLd1cFVIFgAU7NMnnYxDcGQPggYzZtU1D8cbgqsBJbJRBw4pyhkEPPM8zXMk4t2rA+zErgted8T8Dlz2yVoBAAAAAAAAAfQAAAAAAAAAAAAAQwEAAZ7SK3y1EUcyv1FIGawlDKSo5H30JXnCeyV5Gvi36GxsK1QuL5Zet06NGqWC6YxSjpYA59RNnK5PUT9ngIFkyYr9ThBgMwAAAAAAAAABhZzqARa02Xy6hDKTaikX4l4seazLPAl5YmMCbuIcKsQXm8wC3sRr571kjXhez5LhtBMVDeZHlg9ZHgsGkaCexl2ZtvqeNve9u1EiV2KO5B+Wjcufmsmtx6bRrYafOjXZK7uLrrkNyWuiUL0C1uTV/9NDcjTy6EWEkxqfepyv6QYm6BsNRwDrlouLlrls8RlHiZ2MK2MApbOucbUGeobDb3ThyFw6cVAylVzb60W/07YomBRlfkPtLOzsa2OsvbN6NRXDzczkB7StR63GyLGuQ6JpqtMkGGFYytYHOJZ63aq7zXQQDvLJFhD0ZcCndKqXRIkS4ucppX4lni5KsKJe83FJtBqtM0WHGOA/M7lOudb+HwjHryxQRpLm1IGCQZb5Ys0mJVTadNEC3v4nZzPx/OrTKPuSQhpykZ4Gp9RwY9K8mEl1u6FQlqYIrrkd4yZ5Ifn84FS3GGzDaknpsIT//uKWFArzUrHYXV4cAbv94NCoy+oP7ZxBvBkW2POqR0p61Z7eg1TaIJ+yHb48Ap6J9eWn+/Nt8120gBI67G33Mkss+8wHfCyhzJiIJryMaT9H7GwUZnHcnlXB+5LM14V1MmB5oOX7g/w/ezhS6RrhIrNNbHEegRLUAaJxz9Fa2iJybSuIQ4YChd3J21qB0Qs4D6JnDZG7KuxCuD0+UoyXEVgIrp6QQZ5nNeaGSJ+fW3u8VNqnhNDl8wxHspBDvryOUtPus3KiIe3P1dEjPN7Nixs20v5E9bzCbKAOEJSb1mVurOpeHkqXjZ1AFUft3pGo/yjCM2CJZtPXNAOU/t0ZKmMgij9jRc+36fFZlZ3rWEK+fcz5cm589uh+B9jbvo1jqiF2mQAq3kU8XttIPULjhCGCHN+AHiv7Gtq8apP/+CflmVOt5ASBqKBFJLz+07TC99LCVGK5GmK3i9xkN+yisNborCOSOHsYJMHqOujKYGMtt9oAxjNSgbbUItumt1Y96OBBRQddNzK04CI+DuuogarSZ2MUwLsU4kXErHf2Ed3lBo6spwz3IL7VAVuY46y3Y2IBNHSLU5xbKpI+MfuFG5/ot2nKhl88BDj3TeYrBjFiZmVnorMMuQxEuncdWQwT1wJmItg6md/PDHgHtTqWydJfG1rY+igmX569jDMhAkFXuk6wRnPSoAEeSBAssFS0/Zj8qW1HXqH9HdhNsTZAHczjtBC+fFltMsxZi3rus0yyb0JOOCRuF5Fi4w5wdoySnXrkYFU4rRjRMrR642INbn+fntITa4QNk9VN+kTZYf6Iav7JOslN0nzLDNzpYyiv/nauMj+n6YIILT4e9p8J/0wbRznlnV46+Ik8qnNdfguOkRQszMUO87YMvUDBB3RqPr/g9WBmdEPRatEnqlVnrzDp6O6lt9yUlJWQdzPXNcM00h6RP29e+JzBSB3q3YQ9uYEKr5F7uEj0KWbOkrwppnPH/s69OrnQIh08wANTXNMmTOmjcXGOJuSo3k2p39/euW5ispJ722330bbQzkbfQDRI1f8GFTraQ/sIJ9WM2kvKtUY4s8dljPqbUh0Oe5oPlUaLe15w5LQ1qfWtHWMGuUIJYdVlMLfBPRoEo6CrK5toIkmQS5jwZpZq969AqCxJcA2+QmzlSaKNHfVQm10iwOG1k1dxxAtXP8woMAkdy5kDukUnKzaCk3dUqchWmY5IutDC1UaSQeDALS0w23uMkSan+dP3YBdbc4GJUZwMT6bDWbzCtovnd7Nzvoa9g7zqzvI5dUpIDiLn4YzmE35zGtRd+5BWTBRROxtnFhxPBakJiizQHPW0dBY9OEHUj4WFmGdHhDIG59TTmp1vJZDdT/mjUE69ZQJSZhHaU+rR7rzbfhzp3TTJg+JisAUrVf9Aqf8TfWBOFcqCdsoHINQ8IgdCT0oq8mXoY16S3T3yAPRWUUfAqXUvgYdzqhuhK0IHzh6gK0GFDR73/p3nULezYMk3npgCtXamXOZi6QBtSLq6z0wuUhO30V5JfTtPwYOzVKqoBKruUmjg+5qvzkifs3U36qx76gyA3ARgTmiLWphH74BuOgHkUJYmbThBwCeNrrAO42gQYL4OUbLluiDrTlgY1afyglJv9mcjCWflWwxX1FXGRZfZeIoXlBOCwBuDIEpRuLGpOayfLvOU9DO6kW2YQnrdZTkaxsKYjmuM+eIKKBsiFZy5OIOOS42ny35mImj9NS5fOQRzpH2JCw3I7y7JkCwbrvFr7+lazkCw1hO/ms1lPsVs5THTO/AyKcSjR5/WJEiYH6O7XnefDqv+nHDSkoNuXZ4IxuJOGiHTOrd93d3jU32S8gPOaKVKd3dQ7STqK4b0ONqzXQEveFDrw80NFPje07Z71FXG+zQHqx2VZyhtPULf91uGtKyRWYt+7cEt1UEtMux+ZsWcuxWKXfAjI8uJtzMokCwnIIrUcN1BOGQCX808yrz3ujpOWAZSmayMDI9aw4x/o8oLUDGgDW/6aky2Gld0tE47O8OrqJzZOjkfMgbaQ9Q1gVENV0+JkEQjgPwJHwzqJTLreNIau5Z/UfHKMZ1Agk8FRezAprngLiO/KVOPGykkkYcVmwS8MvCT/S6r1BQc8+v8Z+9Vq6XcB0qW1FYWLL0+7471zvHq3aPn2pc7RgL5JClUyiBmck+qkd6ILS/mb142TlqbW/HL6nPb/BelvMhfT4TeKe8aTUd/+cUfOoU64JO9F7EqIW4uhxYUPfya4Q4mkmPTUBXCle+o9bh88R3VvdvzpQnNW/RIPpKQrgTsxoRzA0crw1LhbqO96ltco4LloPYkjup3OXWl4haYFRDJ+Ji9yK96yZ+13Gl3pQ2V5OL5+6ROylrr4zDVvHrBe/TDFRjYR7Jn1VDCFG29gB/yDfUQtlP5V9cbf16xBJ8NsmHcXhjzqsACKfX+osOP5IoXOJadeoPbj9B51/PvaEeCKRYMSW26aGzw6FeqIN4y/9b5NyFMPyXWu0SHtTkyhIS0c4WgteQNQjGouIAFEMH+TfvX5G8Dvo5+TxJiV2SajZll69hTuuKtdih7eOdaQQ+rdFYCHfuP5tcBN8mMJaeAMuv9K8wWMfOVlM7CkFBkWp9e80S3cHJFgTwWgYpJw85R1qt8dBV44kBRtAvh8Bg9b2k+wwAU0nEQKUagOBzQYmAlbFKYMd6OWSLCtMD0Qdz4W9MnyoKweAeX1kF8c0SkOT+Sv39KxcpnjNBDZuFGSTHukqwD79XUvA94W11Wlu+dPHG8N99hTKVYZW1UpNsFYXvxgtFfipy4NrSrdH5jTxWCCK3lzkMrzU5CLYPqsIrk0ERdcJyBCOUNyU5EW7Ces6Pm6z2AjP3PwDbuIyVNAnpIaVz8MdV28SJNohhV0fDcpGa2weEhqS4XPrjOsdDXXVOxRV6uA+tCVDmPa+s6vRysaaJboLTx2v2qHG3U7ENtHS0act0lBNeIbk4YQSHpLpEmtdSfq1ARP4YeGEh4lsYYvceDIOI3Fe7KxDUdGzdG6DQs+MxEm+uWi3jcySSCh2SA+Ot371+er/443mGrSC9ITm9Ew4qtcKxlUkMhBkPMC8W8C6GVdSEP3BLUhxkytjBAw0awktx80pcjXDl7FvowHViBwGVTH9iSY5Lt3oljf2OUh15UPak3a1YiOafCubQaFdqJ8SEiMWi01Svq+QNkCs3oY3ijHWD1KZe2XccWuzi0DwRdP6zhkEnlal751qQ/xnLIEY8dNwYT576SiJMs1fpjYjSk6dyvR7l4PwfKQ4Xct9rOBNS1OCx37CUAW4BVF2yWlGQI5LF28JK8DU3bLiIrzEOpxIZazl/tgXNswI+bj3V1JYnbfCxkKtd3J931gRL1Ca0Th4yz9DxFB47ORq66al/KD78TH16+rLxkRpIysl1jS1Ca8fYrgXmmwbWR+LFmU9++/enWnjRAG8DEOmcAKXhQ8Jm2QUByv+b442dOU4dr4zTv5g6onqAYj6wR0K/dMOQ8FDJECjy+5Us7b4DB1Wm5grGUegMFZseC+ICM7eUijWO5uqlfgDyKUXtwSdg0nTu5Jvk+T/m1Jmy7RVN5OqgKJwlLapv5LSu4I4Xgl6LNA2Ep6ZjIgMkYzGUfGLCv9d3nqR67txE7mol53EvqCz62ZqOb+COGr7qB+nVXLQV8FningTdBzAS9ZnUdyo54xdQFXhO6+YSk5iA+svtQmdj2TWap/innsBXtu6EmNTlKyMpcQSeXCv/c5eNo81qJzTcBq++PNkBxfKexYEBZaJEw+Ou62O+xI14q5SX2g1EX/Tf1809F5S+bCShwQc5n21+l2rYQDQC+eOGmB5+XouDiAIi547Uz5bK2UZP5Nzu3GEswxLjJ8wBvoxAzRIxlnQAA4wVSwjnonAN93tqLaI4PZ9cNhxZc9n6Tmnt8NYHGBVjy7M4Ros36NrUjCwOvrkRgIz6+BCDOcbMpc6h4aFfq21h9ooYKpmDfjpuDinuQJgq2Fex19lEN6oZmmMi1f2vq0KE7QfuWtzwgfEb6hsmMXh6vDnDKoovayfnHdpyoCSCx0ISnZrjaYmvmhmCwJRgrM2larbVerckmCVy1a8ljrlaV56xL+ZL/M6AD9BAahCItAwhKt2vB3lvs87U++RAFWqBuRIgjq+dKITrPX5PZuLnYYHaiQUzHICtpmLc9Q/ax7k9JOqOHt1GlrPp2AtyaAHi2Lznz8yRre2+oacvVmltJM4wJ/dbE9kE+/JSUFPccGU1AnougukrH3DE4qxdhjf8jrk9SpIxd1s5yMB6iiK56nHJyWofKTufVFNj6QAKyGBBFwlrlIojwUnvuu/VDcKu3TRn9Ev+H8dPkm1DqhMSge3o7os+MM9zAIdbPuerNOmt/MY02Eurd4ZsMF5rdPeeHTfdgG/cRAsf2OxsfGafqfxJBG1HmisTT/bPDjaG0HyEqlj3mwpCp+ah8W++0Gx92BZv6JTetSnbuApk0CZLXmtoug55Dyj/L/2sYQAUCLHl03AarKaXmGqHX8Sd3kVG1S+9jw7lZomQSlEQUDU3Bnnb2uTT2vXCzJNvRU+skEqUkxkuR3qhKiw8yyJsaedFORz6F4XNCP3/hlP8a6EOQcoJxlvcYplsfjuO89Q1NxmQVZMkXJuafBdZTUvj7fWXhaXILFDvYJNd0GLkZ9PjGXk+tECKlX+AXvroRNOpLjOqkeoP17YJt6kcjYVImGHC1npialL52UP+KVChZDSN1J4fmsj+tkicAP8BMvcPtU4axkjeGZG+6Peg5s8SGskuMwD1TO22Pmsv5CIhNPiJ8kCIB1jlHMp0WYguyiBZd8tYcih9r/PiYMdK6QOc5nqr1yT39qO6AvP338h+Z7riYLcStJ3DVo/nJTREV4ceVAPjndDxII0mpTJDsbnzekZX8IcT7aRqWkMXnwNjBUc/Gy99TppS1HHAPyy5rm66EHQHSfu+fuUVegkwJjEdxGjr/PWUuo2FQYU1jIYo0q/4ZzWSHV8/+mbrO03OVJqnFXJfHERQHnZ1ShnTbisyYMB1xB0TjlzvYqdODuatBAlfnM2sq/oaYYO+7yUsn2FSLQwEAATaYf2baqSyqphTJ4Vr6SfVsBdBA4zSMX/mZ06JfDkrqdmh7iz4J7mA8GvkMIDgInxlefAUfWmjLUheo22Vh1DP9ThBgMwAAAAAAAAABR496AS2eRbyn6vythAnNihNaeNjGWhu5MdjraSYJjufO+KFGyM5F8dlgSimj/c4nBnmvEKeG1POfch13N4fiVxaD6bziPKEQP7byKK1fBq614RxRv6WONNfLGTaPWDpZUtIywxaYYeF9+wLFMx3wwolbvIjtRS0RhNKATMbux34Ma+ppVGFvjkMlWMbdchilKC8zLlymOEWn0d9V/jM3j2qFwwt46nzvMKNhz5yNh9wX3vDMwrXQ3nfAZq761dXipYCMr8hdPG7kxvr7NiPJCeTdLxJmZANmopfinwM/Qe8+f9glBJPbBjMG+Mxcfzw0ff4ubAxKbru+5BIRRE49nR1NhJnZPtZZqy2mWe6DNmzU9pTz9ZV9K+v5e1JbcikKcQgyIe64cSySe3VPoM67vofHlE7XYttKQsnWfZbrdLif1kVbSENyu64EmK2YGO5aB68nprrDLqQnE0acwt045qWsf59cgSNOgMO3M3s/N/vHuIMyAHWk2GvQ2ty0Yei2IvyDu0qGw/dwtDZjUyhhspc3gsp2pC5gBKpabRIzA9HC5EFtVDXZW5uX2rfT4jnMkKeVwwbMeYqwYRHF1nbghy2F/Tw6stZWYsVKMxn6+2SVds9hnZCQhx44JH31nSMg73eF/+U3Yp+qnvyG9bfd0X6ypmCFvS/28GAMh3+ah7UboHIRX7RywUIOTv/x/PcPRvIJsTd8a7V/X2J8CyZ1Xzx3K3NlAWqYjp65BkrXOzzjKsn0qFTg6wE02EwJCcBHd8cKRLZvzMroM9Jj6QtRbWZ5CJVskSkCB2YoknKl0StI+fgIdfy71r6OeFLzT8Qms121CKO3ncCh2W5kgGYhtyFLZdniXTEly+Fid741e30I65Z1t3WAVMhyzOhj6SJS7Xg9+j8H3AqYFscUqce9DvpCUOuR4RNNgHVwh5pAMUXye2rOhGPnST4nmGvQqD/hy/N1RVeZ+BwdZfmYSbonHnPPYJlxYWgBUus1OyGecRK03clQAqLj1IJhEECM/LkVC8aGmI/tvwBu36HPvaFoz5GtNmjYCQpIYUbu9u9oNMbc5dJAIp0GRcduACYLQCu93gBe2SYA8azFDgj9E0L05xROS+YOJy774fTUYpUebU8j+48KU4LMyaWy0hvgtN99uCfOCeR9Sje7DPQt0uVFvvgGD8MBd5jdasfmAr7NcKn5iVA1HpxzQOkzms2T4O3IJmMRfePLpRTUYQhdqKVKrmyN/S9zIjbG/VT3g6ZcIcrkAOWEcYjDSPeA+zb2n1TtRpB5LVTgvb4NtoogHZfdEFipxCctlj9G9+fQxuI4A/VTHBEtxejgwbZbf2rW2B1vODjQLeOmwhD4n/zuxvNUik8DuCmwqTHocJpu0yBT1TcFBiALZG954TgnU5J3YC/AYMuB4fWGTzlgMmru/9vCBrVILMAR4OElYOjA7jpkEVV9bRe0KEynG+8wz6lcNrNNoHq/KDV4CcCigUYU4jRyp80ytk90SHtHbMW7gFGwuEc3ZOJNBGrNrNdB2lHhOiNGZlpxuDojMJy1ABcE4i77VOW4QJnrBW9LI3FZsPffB8aF3roiJBGV/vK39yAPzWqf1SBOnYCJx5R3apUytwPRMS1LoFne6/lfSnZE0mUpx40nU6NpL0/EhPGfPnCrtrzEqJxvfCTAy+4SuZ1FCwumQ9oK+OsQZTXzf1N+ypkgzjnQ1cXQAcNsFF5/bjpyctIhjDtH+vWOF0r/7m7bGjsaxH2tmct1L622shrwaEdYv8E5I5SaON16I7mJ+iAszpOZGeeUTws+jYMBxfcvVvhwJAFPXtlCKdCBPm6I9WHBYh0b5T2rtEndry0mEmClL4Wiu+mi9CDPAfBVkwGf1Eu9piHKrjBxblSLvSmGk2plRWJf2JAMGIPmt8+ORCxQsaGR/5Z0E0IhantnUQHtswqp8ULnWqNOaZ/iU2acqOE2e0pGF/RREg9EQFbn5UPBW/CqJJFpsLuLCyzipsWtHEcdJHyafHUMn6sgmoYEERyxKt//OcB8VoSnFr5aGifWzR1Z/ZrHRsdM5YrdneOuJJYHT2w/hxVPvTuhqcb+LW9WUhS3JEag8p5qI/z6rvcTWKe0sIZhFvJbejradAYoJw5UzkApL8QrNZcUD6jp12X6iz3Ysnk5ERnzxPr/DwtaHTG68+Hc3eSUNN2MYBPukjmtpHGvC3Qp9oNWGVv+0H9H2ml95WtctnsOAkuaxE+BulHYuv64x97QAYgLhlzx4D6/c0/I1a8bybW4YzVAVDci5d5uQAb3qpU56tXi6vUNNFqfDhqTUZp+fK1u82765XDeiUxZz113Sjm+4/OAKJhFjrGZHLUhrZgNT3n/SMA/0/zvWMpCAI/5ltUiG3oGXIIctKVmCHEDbsd1pXd2P6/lEPFXvOX0hlOc1rpweMgsT6vqV9i4ZOD7Qq3CimKDEgy9elaAqzVkU51c0iJmEry8YspzPKO3Uxhe8TIQa78qCSvJn56NTcq78XPidKBEaGKce7q6u+MTClO4AzAPVUmLZRIJ16ez0I7XLxkHFacPAbdqND9Xkl9actMg5e4uFUMwfNq753eKgib1/kRk7XZVsXfdqMWK4ZhM0NN7wl7SFJkp3RTK5p4T3FKWPQZczCPN1/nmvYQcLtQg7+4/4DsJdrawUX04HYLuxx5dm7y6wCx3Sg7IOi+02CGa6excl6uMsfCnoSjPFYMDnfJSU3z/toY5Dra1b+Qj1LJxkEyI2tpf2QGxfvh2oajI/V+scwRaNQd7EzNzgkCsusZKsrzHntlY2+ndxPQy8FQT19I7irC3fbM8o6pn0KIEXQiQILr7yQK1krWwtVRPMKqXoROzYtud7H8ksanMcqO4Q9gj2l6nXcVY2XTQnm6BVrxVLcha8col5CWglxynsXUJLccBAM1QyGgPKImGCQ0R2eUMRGtBs6ynEuSkSEVepgBMeru99xhFi/StAqfI5sAbVgwPflSCot+jYxl3Ef47DixoPi24e7PeLiRvSNbeIEW+p5lk5zUgbwA2YTFcza3hoU5WeqhPB7qMopxUSc7qPVUf8DAHQOUIm7frlLQ5G2ZaqRhTsDKkXwUZAYCkk4QhS/2gonL/We/S/cQwcgOfQdWJkI+fSP0fV6n5DRtMyfRdC5+E/Lr1pSBnzdacTCDTDfYs8ekoAiWql0Dc1Dc8qmlz/TSmm1klM1LnACpY/Mrv7Df+OuvYiYoEX/qpTDjGuAj2BiX8zkKBNN9MmzFbVh6dCKa4Huguh4dNw+zSDNQcHzKTKOzTph0jcD99WhRiei+UoAae0MFTnyQqmB9Z0wJg+Unyu0g4UBd5M0BGsbKiOUJZDbO/uNpIQ8QeCevuQJ1UTK4WOvKJVrIAKE3bxpBCg29xflvV7tpLT/rygwBdfDwc0ThjldvV9ap6oYoi29wyfdS1dvFexqQRW1kEkkYkRCNYwJ2NfRZGz2RvDMUWz1giwodLmc1kHGnS8vbHO0rKPwLMhx88Yf5owI1LJ7KMYIx5AvmOgIj1qA3yuCVMr/RkDdcZmUoT8+dhjsKOli7ajWF4WeSEJR3Uz1cQGxD2e98Vj3IbiNPcWgV8hkpebz9W+uUkzIBuyKM02eHbTOUgD/hMeF15OGdkAqGfXyKzv/v2Q+h49bQGNiSP5tLWOQs9RYtYbP3vm40W0iifbm3b3W1PLJ3jYmTeS3aW2K0VOMsHPzZhc1qnm/1r1uN2inrC21SzmlF7sDAH+7y5owik1RkKEDNCTwYTl1zHnDGSYc2saQWdrpkMOHY+/H2MJcYdP/bpM7FYnq/7C0vH0uaQ/KAnR0SOIqk9trSF5Y/hzFEWNugQawYywBZFZtghXT7xBdZWmnxRi3BExGM7Ud+BrhAJc0mYZTJsHLQmLr0+60EN4um3mS1+JHOnMBzB9+5NqErvbGy4N8OCIFS94PRR0lka+/np8PqXvdGlnJwiTMjLH6t3J4NBG4SGUmgTElMyubrgyL3L+shJsj+boRWLW+vOQIP1NXPXuy+MWsknGhFt4ehPpGQuU/AeLzX/v4aL18YaKiJMpY9TvuOjIZVs4ybF0Okl+GW/+zpQoWlKC796mDB1w0GwecW6raa8YxxGh6mT6l7RoXkXjWAO7aaT9ncmSwCZjH7fOoe7n0Ir+yWlpzX2BHeJgz5KT01Hq8Dg0yJQVi6iVjxDoxCWRUz7VIASDPASmbyS6I7OsZFE1zyKPp3qs9gPNgCp6msToKkrHg2+YLy9519MTslx8qN6AyrORhOkq9BA/jlICQPzUs2mZBueDM4Jn8c9anMEQkeJncYwu7rInmOwcQQmT7U7+SiUt2ZWxNbxZ1wbxWeBq3Mb8ui2ZALTCbTmAtZQfieyHAb44s2odYHqJtPUiFoinpD6HBnc3KY8ew4PDZf0HI0UyCiY9Mkv079D5oB/8TptRkEBIUfAPJfVsg9NoN7kKqQ7WBSdc8VdDo+QcO/qBxPaMyr1AuHewtX40Iwp41D8yzTzmIzZxEtYluliyl0sgwJS4rE1IoCRV+eLVpSOSBRRzLMqtNtwVUcAdqWSWbiy7butqMQ8JiErdlEs6Nzw3RAXaEFuOpVjcoYwcFdRTgYZFi9T5B3tg6vIVG3TpfTnf8L3QL35v3Qee2VWfVppOEYNE1mBetFuMnNY8lI7aY1uXA8gN0FfeBLEC0aIMm6nx4PkJfNv26yUaRvd/W9ilEfXUHWpmSyc8H6RPc3y9QI+2Hv8OCnGShykMWI52BocuBJtZkforp7vf1phddxM/zAhqAbkH8SDKV5GBpcO8hAfH5/yS9VO7WCUYCrETrcpJxgoojpLktegjsbja97F17LSTzgjdF/4DK1kxPloWqm0EpWN29tVv3SdeFBq+C8oH9JGVVYiVPC+erWDu/lk/wT4q8gftcdfe1c70meHPvZLJTRSjz1oppxCIgnhqcwjKUqi/9gdXYkwvqbSVQH6Bg1Y6Lzd3nmC16Y69fonHWJkgZXThL7iGQiVIfc6EbXRPjghZ/uwGPBXNtjl40xadW+XqQCYMfddDG0bRxXigU/FZeljZo6vWpXeHofNbogC4bAmdi+NqQpbLnRLbwLWV8RfG+RQcdAwhBwcuf6MxgQzrSAK7FsgNCXkCcmkS2peKkAVv0nvq6NhAxoqBcmYaMgjvQKmgsiEc4O2y276KrEfXSnRE9OViprMGqjfQkN6+t948G4+uFL+RfsHQtLXXTND3ufQLbXyvFg+Y8ZNCUdCp9jPjpFYHyA2T07SS1y0DesdrFaNNM197HKvPXs4i7i7sJ9aAFgchDAcCy7Q4DQvIUiLFTtIJNbdGRI8dNqpjeSj4z5YQsfW6rPZ1DBKf05R7jwpXWdAv6dMcu/bnycIhPcAqaKP5n8U3grF6oN7OA7szAp24UXFYkAtHzhgFtJ3etSDRCgpLUv0c32uYDKchwUQaMSY6nxML0FqTNzHrAH1UpYAcz4iYhAtgQsuGS26UZ4YnZuFuj+4hWSMIE7rQfmQvevlrSykbE+rN5M4jfTrQDb5y3B7MWHEiM8opoPeGW4/AAAAAQF6Cx8a2BlkctPjHz4iViq3XyIJE3XAucNvCYxfvwAqczojCFYcb1IVznfj3Zm04ZI6XjT52c1Ktwi/gCAS7UvlsCp7AlDf+FTWeFbkESg8KQW2RfD7/9QG8JMkUbPGknDlRPagFgAU7NMnnYxDcGQPggYzZtU1D8cbgqsBCGwCSDBFAiEAyPC/qY9/Dn1P5z1L7THSqenKiiGXnzcQWDEtB5xlrmgCIDGLK9fPfQ8pPR3xrfhxX1bSAgqYfeuxKspNGUw6uVRcASECmNKCWvM2zEotKuKJQGHqhRYeJfcBww97C2+xX3O10B8AAAAA
```

**Comment by tiero** [comment]
Are the two PSET different? One is base64 and the other hex string?

**Comment by altafan** [comment]
my bad, just edidted the description.

**Comment by altafan** [comment]
After looking at the [`EncodeBase64`](https://github.com/ElementsProject/elements/blob/master/src/util/strencodings.cpp#L128) function of the elements implementation, it seems to me that to whole partial transaction is compressed, rather than just the range proofs.
The `ConvertBits` function used there should be the same used for blech32, so we already have it in out lib and in our pset package maybe it would just be enough to call it before returning the encoding partial transaction in order to shrink it.

**Comment by louisinger** [comment]
@altafan it appears that the transaction you created with the CLI is unblinded (and does not contain rangeproof) did you use the rpc `blindpsbt` ? If I create a PSET and then blind it using elments CLI i get a larger result: https://gist.github.com/louisinger/f99064ecaa9545ec57937079216a0c7b

_As you can see, after `blindpsbt` the string length ~= 13000. The go-elements transaction you created ~= 12000._

However I notice that the way data are encoded is a bit different than go-elements do. For example there is no range proof or surjection proof in witnessUtxo. Instead, we have the value `asset_blinding_factor` and `value_blinding_factor`. 

In conclusion, I think that CLI's encoding and Vulpem's encoding are different. From there, I think that different results can be expected. I do not understand what exactly need to be fixed ? do we need to refacto the way we store proofs and blinding factors in `go-elements` to fit the Elements CLI method ?

**Comment by tiero** [comment]
cc @altafan

**Comment by altafan** [comment]
I decoded the pset with the CLI and saw that partial outputs are full of data, while when we parse ours with the CLI, partial outputs are always empty. Also, the unsigned transaciton still has unblinded values, while we override them when blinding.

Our pset format is definitely not the same as the Elements one. But neither their format prevent the base64 to "explode", apparently.
So what I see is that Elements, first of all, doesn't alter the unsigned tx until the finalization of the partial transaction. Given this, for every output, it saves in the partial output the blind pubkey, blind factors, commitments, nonces and proofs. Basically all the confidential values are stored in the partial outputs fields and wrote in the UnsignedTx only when finalizing the tx... like it happens for the signatures that are stored in the partial inputs.

**Comment by altafan** [comment]
@louisinger can you provide the steps for blinding a pset with Elements CLI so that we can reproduce your test case?

**Comment by louisinger** [comment]
Script to reproduce blinded psbt using the elments-cli:
```
TXID=$(ecli listunspent | jq -r '.[0].txid')                                                                                                                                    
VOUT=$(ecli listunspent | jq -r '.[0].vout')
UNBLIND=$(ecli walletcreatefundedpsbt "[{\"txid\":\"$TXID\",\"vout\": $VOUT}]" "[{\"AzpxT39iGzGzruc3MMjN7LGYnM2ft1WKsKa5mDoQNrGigGgzvRa61eUWCiaQ1K88yZPZtEfy2a7Sfm3M\":\"100\"}]" | jq -r '.psbt')
ecli blindpsbt $UNBLIND
```
